### PR TITLE
feat(firestore): query operators: 'not-in' & '!='

### PR DESCRIPTION
**Pull request by russellwheatley**
### Description

Feature update for Firestore; where query operators:
 - [not-in](https://firebase.google.com/docs/firestore/query-data/queries#in_not-in_and_array-contains-any)
 - [!=](https://firebase.google.com/docs/firestore/query-data/queries#not_equal_)

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- My change supports the following platforms;
  - [x] `Android`
  - [x] `iOS`
- My change includes tests;
  - [x] `e2e` tests added or updated in `packages/\*\*/e2e`
  - [ ] `jest` tests added or updated in `packages/\*\*/__tests__`
- [x] I have updated TypeScript types that are affected by my change.
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Star this repo on GitHub ⭐️
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter

### CODE REVIEW

File: packages/firestore/ios/RNFBFirestore/RNFBFirestoreQuery.m [hunk 1]
```diff
@@ -58,6 +58,8 @@ - (void)applyFilters {
 
     if ([operator isEqualToString:@"EQUAL"]) {
       _query = [_query queryWhereFieldPath:fieldPath isEqualTo:value];
+    } else if([operator isEqualToString:@"NOT_EQUAL"]) {
```
Comment (mikehardy) [inline packages/firestore/ios/RNFBFirestore/RNFBFirestoreQuery.m:61]: Space after if here is the only thing I can see. This is sweet!

File: tests/ios/Podfile.lock [hunk 2]
```diff
@@ -1091,25 +1091,25 @@ SPEC CHECKSUMS:
...
+  RNFBPerf: 0c08e45726f7a19487b79cef3d12ee7e917c8b7a
+  RNFBRemoteConfig: 85df9235f46c20e293257b6e481412ac585e2966
+  RNFBStorage: 72404d4977261c0d7060e87c3d0cf7f7e060c6a3
   Yoga: 3ebccbdd559724312790e7742142d062476b698e
 
-PODFILE CHECKSUM: 555dacf747a9f802fb0d5efa4c6f558ca6129c4d
+PODFILE CHECKSUM: 2b670e97319c5057f900292e8500c4c38d83aa3c
 
-COCOAPODS: 1.10.0.rc.1
+COCOAPODS: 1.9.3
```
Comment (mikehardy) [inline tests/ios/Podfile.lock:1115]: This is unrelated just a tangent that 1.10 is out as general release now.

**Review by mikehardy — DISMISSED** [review]
A whitespace nit but this looks great

**Comment by mikehardy** [comment]
One lint thing about whitespace too apparently?

File: packages/firestore/ios/RNFBFirestore/RNFBFirestoreQuery.m (hunk 1 above)
Comment (russellwheatley) [reply packages/firestore/ios/RNFBFirestore/RNFBFirestoreQuery.m]: good catch 👍

File: tests/ios/Podfile.lock (hunk 2 above)
Comment (russellwheatley) [reply tests/ios/Podfile.lock]: updated 😄

File: packages/firestore/ios/RNFBFirestore/RNFBFirestoreQuery.m (hunk 1 above)
Comment (mikehardy) [reply packages/firestore/ios/RNFBFirestore/RNFBFirestoreQuery.m]: I am *ace* at whitespace haha. Does not quite make up for objective-c being near worthless but we'll just skip that

**Comment by mikehardy** [comment]
Just failing yarn install on CI E2E (seriously? yarn install is going to flake, c'mon...). I'll herd it through, I'm excited to see this go in

**Comment by mikehardy** [comment]
Success!
Locally I temporarily disabled the auth package in mocha.opts so I could get past it, and I ignored the firestore emulator per note here: https://github.com/invertase/react-native-firebase/pull/4133#discussion_r515357208

And everything passed :-)
